packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -81,39 +70,112 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
       assert(key == this);
 
       final Uri resolved = Uri.base.resolve(key.url);
-      final HttpClientRequest request = await _httpClient.getUrl(resolved);
-      headers?.forEach((String name, String value) {
-        request.headers.add(name, value);
+
+      final Completer<TransferableTypedData> bytesCompleter = Completer<TransferableTypedData>();
+      final RawReceivePort downloadResponseHandler = RawReceivePort((_DownloadResponse response) {
+        if (response.bytes != null) {
```

Comment:
It's possible for the completer to have already completed with an error here, as would be the case if an uncaught error happened in our worker isolate, triggering our error listener, which would send an error message over this port.

I think you can add a check for `isCompleted` here:

```dart
if (bytesCompleter.isCompleted) {
  // If an uncaught error occurred in the worker isolate, we'll have
  // already completed our bytes completer.
  return;
}
```

Reply:
Done

Reply:
Um, not sure. Any reason why this would be preferable?